.github/workflows/ci.yml

```
@@ -0,0 +1,58 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+
+jobs:
+  linting:
+    name: "Run linters"
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout code
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+      - name: Install dependencies
+        run: |
+          make venv
+          pip install -r requirements-dev.txt -e .
+
+      - name: Run linter
+        run: |
+          black infra_event_notifier tests
+          flake8 infra_event_notifier tests
+  typing:
+    name: "Type checking"
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout code
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+      - name: Run mypy
+        run: |
+          make venv
+          make typecheck
+
+  tests:
+    name: "Run tests"
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 3
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout code
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
```

```suggestion
          python-version: |
            3.11
            3.12
```
we should match the versions we build for: https://github.com/getsentry/infra-event-notifier/blob/main/.github/workflows/build.yml#L18-L20